### PR TITLE
Enforce formatting of TOML with `taplo`

### PR DESCRIPTION
**Pull request by ErichDonGubler**
**Connections**

\-

**Description**

Title says all.

**Testing**

CI!

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] ~~Add change to `CHANGELOG.md`. See simple instructions inside file.~~ Not necessary.

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
An example failure for this additional check: https://github.com/gfx-rs/wgpu/actions/runs/11182596736/job/31089235852?pr=6366#step:6:26

File: rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Wumpf) [inline rustfmt.toml:1]: if we don't config anything, why do we have this file even

**Review by Wumpf — APPROVED** [review]
yes please! :)

File: rustfmt.toml (hunk 1 above)
Comment (cwfitzgerald) [reply rustfmt.toml:1]: I'm going to be extremely annoying and suggest a config that matches what I was trying to do manually

```
[formatting]
align_comments = true
indent_string = "    "
reorder_arrays = true
reorder_inline_tables = true
reorder_keys = true
```

Comment (Wumpf) [reply rustfmt.toml:1]: 🤷 pr to demo what these look like practically ;)

Comment (cwfitzgerald) [reply rustfmt.toml:1]: https://github.com/gfx-rs/wgpu/pull/6376

**Comment by nical** [comment]
Please document somewhere visible (the readme and the PR template?) what people need to do locally to satisfy the new constraint.

File: rustfmt.toml (hunk 1 above)
Comment (ErichDonGubler) [reply rustfmt.toml:1]: @cwfitzgerald: I've moved the addition of a `taplo.toml` with the indentation change into this PR. This minimizes diffs. in this PR and #6376. This movement also narrows the focus of that PR on setting `reorder_*` config. to `true`, which I think is a nice, contained change that we can merge after this. 🙂

**Comment by ErichDonGubler** [comment]
@nical: The `README` doesn't mention anything about a developer environment, but #5680 sure should. I've added content about setting up Rust and Taplo there in 07f324bbbf41, and also included a reference to `taplo` in the PR template here with 7d9a95b.

**Comment by ErichDonGubler** [comment]
Just waiting for @gfx-rs/naga to approve here.

**Review by teoxoy — APPROVED** [review]

**Comment by ErichDonGubler** [comment]
Heads up, @crowlKats, this is coming your way and it affects Deno (tho in a minor way). 🫡

**Pull request merged**